chainercv/links/ssd/ssd.py

```diff
@@ -0,0 +1,257 @@
+        'initial_bias': initializers.Zero(),
+    }
+
+    def __init__(self, n_class, **links):
+        self.n_class = n_class
+
+        super(SSD, self).__init__(
+            loc=chainer.ChainList(),
+            conf=chainer.ChainList(),
+        )
+        for name, link in six.iteritems(links):
+            self.add_link(name, link)
+
+        for ar in self.aspect_ratios:
+            n = (len(ar) + 1) * 2
+            self.loc.add_link(L.Convolution2D(
+                None, n * 4, 3, pad=1, **self.conv_init))
+            self.conf.add_link(L.Convolution2D(
+                None, n * (self.n_class + 1), 3, pad=1, **self.conv_init))
+
+        # the format of default_bbox is (center_x, center_y, width, height)
+        self._default_bbox = list()
+        for k in range(len(self.grids)):
+            for v, u in itertools.product(range(self.grids[k]), repeat=2):
+                cx = (u + 0.5) * self.steps[k]
+                cy = (v + 0.5) * self.steps[k]
+
+                s = self.sizes[k]
+                self._default_bbox.append((cx, cy, s, s))
+
+                s = np.sqrt(self.sizes[k] * self.sizes[k + 1])
+                self._default_bbox.append((cx, cy, s, s))
+
+                s = self.sizes[k]
+                for ar in self.aspect_ratios[k]:
+                    self._default_bbox.append(
+                        (cx, cy, s * np.sqrt(ar), s / np.sqrt(ar)))
+                    self._default_bbox.append(
+                        (cx, cy, s / np.sqrt(ar), s * np.sqrt(ar)))
+        self._default_bbox = np.stack(self._default_bbox)
+
+    def to_cpu(self):
+        super(SSD, self).to_cpu()
+        self._default_bbox = chainer.cuda.to_cpu(self._default_bbox)
+
+    def to_gpu(self):
+        super(SSD, self).to_gpu()
+        self._default_bbox = chainer.cuda.to_gpu(self._default_bbox)
+
+    def features(self, x):
+        """Compute feature maps from a batch of images.
+
+        This is a virtual method.
+        The inheriting class must implement this method.
+
+        Args:
+            x (ndarray): An array holding a batch of images.
+                The images are preprocessed by :meth:`prepare` if needed.
+
+        Returns:
+            list of Variable:
+            Each variable contains a feature map.
+        """
+        raise NotImplementedError
+
+    def _multibox(self, xs):
+        ys_loc = list()
+        ys_conf = list()
+        for i, x in enumerate(xs):
+            loc = self.loc[i](x)
+            loc = F.transpose(loc, (0, 2, 3, 1))
+            loc = F.reshape(loc, (loc.shape[0], -1, 4))
+            ys_loc.append(loc)
+
+            conf = self.conf[i](x)
+            conf = F.transpose(conf, (0, 2, 3, 1))
+            conf = F.reshape(
+                conf, (conf.shape[0], -1, self.n_class + 1))
+            ys_conf.append(conf)
+
+        y_loc = F.concat(ys_loc, axis=1)
+        y_conf = F.concat(ys_conf, axis=1)
+
+        return y_loc, y_conf
+
+    def __call__(self, x):
```

Comment:
Like this
https://github.com/yuyu2172/chainercv/blob/d4f0188db8296efca6ec67e6d3f3de40f8c78e8e/docs/source/reference/links.rst